sway/desktop/xdg_shell.c

```diff
@@ -291,9 +291,8 @@ static void handle_commit(struct wl_listener *listener, void *data) {
 		wlr_xdg_surface_get_geometry(xdg_surface, &new_geo);
 		struct sway_container *con = view->container;
 
-		if ((new_geo.width != con->content_width ||
-					new_geo.height != con->content_height) &&
-				container_is_floating(con)) {
+		if ((new_geo.width != con->surface_width ||
+					new_geo.height != con->surface_height)) {
 			// A floating view has unexpectedly sent a new size
```

Comment:
I don't think this is specific to floating views anymore so the comment should be updated (repeat for `xdg_shell_v6` and `xwayland`)

Reply:
Fixed.